vms/platformvm/state/validator_metadata.go

```diff
@@ -4,36 +4,56 @@
 package state
 
 import (
+	"math"
 	"time"
 
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/codec/linearcodec"
 	"github.com/ava-labs/avalanchego/database"
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/utils/constants"
 	"github.com/ava-labs/avalanchego/utils/set"
 	"github.com/ava-labs/avalanchego/utils/wrappers"
-	"github.com/ava-labs/avalanchego/vms/platformvm/genesis"
-	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
 )
 
 // preDelegateeRewardSize is the size of codec marshalling
 // [preDelegateeRewardMetadata].
 //
 // CodecVersionLen + UpDurationLen + LastUpdatedLen + PotentialRewardLen
-const preDelegateeRewardSize = wrappers.ShortLen + 3*wrappers.LongLen
+const (
+	preDelegateeRewardSize = wrappers.ShortLen + 3*wrappers.LongLen
 
-var _ validatorState = (*metadata)(nil)
+	v0tag                    = "v0"
+	validatorMetadataCodecV0 = uint16(0)
+)
+
+var (
+	_ validatorState = (*metadata)(nil)
+
+	validatorMetadataCodec codec.Manager
+)
+
+func init() {
+	c := linearcodec.New([]string{v0tag}, linearcodec.DefaultMaxSliceLength)
+	validatorMetadataCodec = codec.NewManager(math.MaxInt32)
+
+	err := validatorMetadataCodec.RegisterCodec(validatorMetadataCodecV0, c)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Not sure I want to do that. I think keeping in the very same file suggests that this codec ought to be used to validatorMetadata only? (note that in continuous staking I will have to add a delegatorMetadata and use the very same codec)

Reply:
I dont know if there is any implication. I wouldnt ask if this was a plain struct, but the init func looks a bit weird. Nothing really serious though.

Reply:
I use the init function to initialize this code the same way we do with txs.Codec or blocks.Codec. The init is executes before any other code, so that we are guaranteed that codecs are well initialized when node starts